src/Controller/EnMarche/VotingPlatform/VotingPlatformController.php

```diff
@@ -0,0 +1,157 @@
+    private $candidateGroupRepository;
+
+    public function __construct(
+        RedirectManager $redirectManager,
+        VoteCommandStorage $storage,
+        VoteCommandProcessor $processor,
+        CandidateGroupRepository $candidateGroupRepository
+    ) {
+        $this->redirectManager = $redirectManager;
+        $this->storage = $storage;
+        $this->processor = $processor;
+        $this->candidateGroupRepository = $candidateGroupRepository;
+    }
+
+    /**
+     * @Route("", name="_index", methods={"GET"})
+     */
+    public function indexAction(Election $election): Response
+    {
+        $voteCommand = $this->storage->getVoteCommand($election);
+
+        if (!$this->processor->canStart($voteCommand)) {
+            return $this->redirect($this->redirectManager->getRedirection($election));
+        }
+
+        $this->processor->doStart($voteCommand);
+
+        return $this->renderElectionTemplate('voting_platform/index.html.twig', $election);
+    }
+
+    /**
+     * @Route("/vote", name="_vote_step", methods={"GET", "POST"})
+     */
+    public function voteAction(Request $request, Election $election): Response
+    {
+        $voteCommand = $this->storage->getVoteCommand($election);
+
+        if (!$this->processor->canVote($voteCommand)) {
+            return $this->redirectToElectionRoute('app_voting_platform_index', $election);
+        }
+
+        $this->processor->doVote($voteCommand);
+
+        $form = $this
+            ->createForm(
+                $this->getCandidateFormType($election),
+                $voteCommand,
+                ['candidates' => $candidateGroups = $this->candidateGroupRepository->findForElection($election)]
+            )
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->processor->doConfirm($voteCommand);
+
+            return $this->redirectToElectionRoute('app_voting_platform_confirm_step', $election);
+        }
+
+        return $this->renderElectionTemplate(
+            sprintf('voting_platform/vote_step/%s.html.twig', $election->getDesignationType()),
+            $election,
+            [
+                'form' => $form->createView(),
+                'candidate_groups' => $candidateGroups,
+            ]
+        );
+    }
+
+    /**
+     * @Route("/confirmer", name="_confirm_step", methods={"GET", "POST"})
+     */
+    public function confirmAction(Request $request, Election $election): Response
+    {
+        $voteCommand = $this->storage->getVoteCommand($election);
+
+        if (!$this->processor->canConfirm($voteCommand)) {
+            return $this->redirectToElectionRoute('app_voting_platform_vote_step', $election);
+        }
+
+        $form = $this
+            ->createForm(ConfirmActionType::class)
+            ->remove('deny')
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid() && $form->get('allow')->isClicked()) {
```

Comment:
```suggestion
        if ($form->isSubmitted() && $form->isValid()) {
```
don't seems to be needed as you remove the deny button before

Reply:
fixed, tnx